ml-agents/mlagents/trainers/learn.py

```diff
@@ -159,6 +162,7 @@ def run_training(run_seed: int, options: RunOptions) -> None:
         env_manager.close()
         write_run_options(write_path, options)
         write_timing_tree(run_logs_dir)
+        StatsReporter.save_state(os.path.join(run_logs_dir, "training_status.json"))
```

Comment:
I don't think this is the best place to store this global state. I would put it within the save model calls.

Reply:
Hmm, that makes sense since we'd want to resume on the event of a crash as well. 

However the save model calls will be moved to within the trainers, since it seems logical that each trainer drives its own checkpointing (at possibly different frequencies). Wouldn't want the global save to be there as well. We could have each trainer manage its own save state, or have every trainer just trigger the save_state function.

An alternative would be to just write to the JSON every time a new state is written. For stuff like the lesson number (which is very infrequently written) this is OK.